From 6b7a3a917e1d231eecf55226922373ce038255ff Mon Sep 17 00:00:00 2001
From: "Ye.Li" <B37916@freescale.com>
Date: Mon, 17 Aug 2015 14:24:56 +0800
Subject: [PATCH] MLK-11367 imx: mx6ul: Add cma bootargs for 6ul 9x9 board

Since the i.MX6ul 9x9 evk board only has 256MB LPDDR2, while the
CMA size used in kernel configuration is 320MB, so we have to set
another value for the 9x9 evk board.

This patch sets the cma=96M bootargs to in uboot. So the kernel will
overwrite to use the new value.

Signed-off-by: Ye.Li <B37916@freescale.com>
---
 include/configs/mx6ul_14x14_evk.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/configs/mx6ul_14x14_evk.h b/include/configs/mx6ul_14x14_evk.h
index 1d82b04..ba2130b 100644
--- a/include/configs/mx6ul_14x14_evk.h
+++ b/include/configs/mx6ul_14x14_evk.h
@@ -127,6 +127,7 @@
 #if defined(CONFIG_MX6UL_9X9_LPDDR2)
 #define CONFIG_DEFAULT_FDT_FILE "imx6ul-9x9-evk.dtb"
 #define PHYS_SDRAM_SIZE			SZ_256M
+#define CONFIG_BOOTARGS_CMA_SIZE   "cma=96M "
 
 /* PMIC */
 #define CONFIG_POWER
@@ -136,6 +137,7 @@
 #else
 #define CONFIG_DEFAULT_FDT_FILE "imx6ul-14x14-evk.dtb"
 #define PHYS_SDRAM_SIZE			SZ_512M
+#define CONFIG_BOOTARGS_CMA_SIZE   ""
 #endif
 
 
@@ -167,6 +169,7 @@
 
 #define CONFIG_MFG_ENV_SETTINGS \
 	"mfgtool_args=setenv bootargs console=${console},${baudrate} " \
+	    CONFIG_BOOTARGS_CMA_SIZE \
 		"rdinit=/linuxrc " \
 		"g_mass_storage.stall=0 g_mass_storage.removable=1 " \
 		"g_mass_storage.idVendor=0x066F g_mass_storage.idProduct=0x37FF "\
@@ -187,6 +190,7 @@
 	"console=ttymxc0\0" \
 	"bootargs=console=ttymxc0,115200 ubi.mtd=3 "  \
 		"root=ubi0:rootfs rootfstype=ubifs "		     \
+		CONFIG_BOOTARGS_CMA_SIZE \
 		"mtdparts=gpmi-nand:64m(boot),16m(kernel),16m(dtb),-(rootfs)\0"\
 	"bootcmd=nand read ${loadaddr} 0x4000000 0x800000;"\
 		"nand read ${fdt_addr} 0x5000000 0x100000;"\
@@ -210,6 +214,7 @@
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
 	"mmcautodetect=yes\0" \
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
+	    CONFIG_BOOTARGS_CMA_SIZE \
 		"root=${mmcroot}\0" \
 	"loadbootscript=" \
 		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
@@ -233,6 +238,7 @@
 			"bootz; " \
 		"fi;\0" \
 	"netargs=setenv bootargs console=${console},${baudrate} " \
+	    CONFIG_BOOTARGS_CMA_SIZE \
 		"root=/dev/nfs " \
 	"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" \
 		"netboot=echo Booting from net ...; " \
-- 
1.8.0

